src/common/actions/snaps.js

```diff
@@ -12,14 +12,16 @@ export const REMOVE_SNAP = 'REMOVE_SNAP';
 export const REMOVE_SNAP_SUCCESS = 'REMOVE_SNAP_SUCCESS';
 export const REMOVE_SNAP_ERROR = 'REMOVE_SNAP_ERROR';
 
-export function fetchUserSnaps(owner) {
+export function fetchUserSnaps(owner, showFetching = true) {
```

Comment:
Another option is that the UI always shows the isFetching state but in a less obtrusive way, for example if the spinner is behind the content (only visible when we have no content). Cheating, I know.